### PR TITLE
Support Windows testing for Kokoro.

### PR DESCRIPTION
**Pull request by 95-martin-orion**
Adds Windows-specific options to BUILD rules.

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
Kokoro doesn't appear to trigger on draft PRs. Marking "ready for review" in an attempt to trigger Kokoro, but this PR is still a work in progress.

**Comment by 95-martin-orion** [comment]
> Kokoro doesn't appear to trigger on draft PRs. Marking "ready for review" in an attempt to trigger Kokoro, but this PR is still a work in progress.

This isn't necessarily true - I don't see Kokoro builds on _any_ PRs now - including ones where Kokoro previously ran! Something strange has happened. I'll look into this.

**Comment by 95-martin-orion** [comment]
> ...Something strange has happened. I'll look into this.

This appears to be tied to use of the `kokoro:run` label: only PRs with this label will trigger Kokoro. It's still not clear to me why old runs are no longer visible, but that's not too urgent at the moment.

**Comment by 95-martin-orion** [comment]
The Kokoro Windows build is now passing. Currently it only runs tests, as apps aren't a blocker for TFQ integration; for completeness I've added a TODO to also get the apps working in Windows.

It's likely _possible_ to build in Windows in the current state, but for TFQ integration there are some additional constraints that we're required to meet.

File: tests/BUILD [hunk 1]
```diff
@@ -4,9 +4,23 @@ sse_copts = ['-msse4']
 omp_copts = ['-fopenmp']
 omp_linkopts = ['-lgomp']
 
+windows_copts = [
+    "/arch:AVX",
+    "/std:c++14",
```
Comment (sergeisakov) [inline tests/BUILD:9]: What is the reason to use C++14 instead of C++11?

Comment (95-martin-orion) [reply tests/BUILD:9]: C++14 is also used in the TFQ builds, which are what we need to align with for these tests. I'll double-check these with their team to be certain this was intended, but for now we'll stick with this.

(As an aside, and possibly a note to my future self: `/std:c++11` is not a recognized flag when building with Kokoro, as it uses C++11 by default.)

**Pull request merged**